src/Validator/EditorValidator.sol

```diff
@@ -3,33 +3,52 @@ pragma solidity ^0.8.13;
 
 import {IValidator} from "./IValidator.sol";
 
-// set this to wiki as validator and it will check if user has editor NFT and has not more than 5 edits in last 24h . number of edits is settable
 contract EditorValidator is IValidator {
-
     /// -----------------------------------------------------------------------
     /// Errors
     /// -----------------------------------------------------------------------
 
     error WrongIPFSLength();
+    error ExceededEditLimit();
 
     /// -----------------------------------------------------------------------
     /// Storage variables
     /// -----------------------------------------------------------------------
-
-    /// @notice The amount of edits
-    mapping(address => uint256) public edits;
+    mapping(address => uint256[]) edits;
+    uint256 count;
 
     /// -----------------------------------------------------------------------
     /// External functions
     /// -----------------------------------------------------------------------
 
     // mapping of editors and edits / blocktimestamps in order of calculate how many edits in X period of time
-    function validate(address _user, string calldata _ipfs) external returns (bool) {
-        edits[_user]++; // TODO: dummy code
+    function validate(address _user, string calldata _ipfs)
+        external
+        returns (bool)
+    {
+        // check if array is full
+        if (edits[_user].length < 5) {
+            edits[_user].push(uint256(block.timestamp)); // add new element to array
+        } else if (edits[_user].length == 5) // check if array is full
+        {
+            // check if the oldest edit time is older than 1 day, #NOTE: the oldest edit time is the first element in the array
+            if (block.timestamp - edits[_user][0] >= 1 days) {
+                // shift the first element to the end of the array and reorder the array
+                for (uint256 i = 0; i < edits[_user].length - 1; i++) {

```

Comment:
edits[_user].length--; it is read-only and can't be used to resize

Reply:
the reason for the current implementation is that if I remove the value from the array with delete it will leave a gap

Reply:
> the reason for the current implementation is that if I remove the value from the array with delete it will leave a gap

yes, i guess its fine. I am a not big fan of iterating over arrays on contracts but I guess its needed in this case. 

Something you can test its to get in memory that array in the beginning of the function and replace it in the end if needed, that way you won't be accesing storage functions. 

Pls just for fun do a `forge snapshot` before and after the change to see how it affects 👍🏻

Reply:
`        uint256[] memory temp = edits[_user];
`

I tried this, but I got an error of `Member "push" is not available in uint256[] memory outside of storage.` when trying to push
`ed.push(uint256(block.timestamp));`.

if I use `storage` instead it worked but the gas is higher `uint256[] storage temp = edits[_user];`